azurelinuxagent/common/protocol/restapi.py

```diff
@@ -161,6 +176,24 @@ def __init__(self,
         self.dependencyLevel = dependencyLevel
         self.dependencies = [] if dependencies is None else dependencies
 
+    def resolve_dependencies(self, ext_handlers):
```

Comment:
More "pythonic":
```
def resolve_dependencies(self, ext_handlers): 
    resolved_dependencies = [] 
    for (handler_name, ext_name) in self.dependencies: 
        # Gracefully handle garbage cases where handler name does 
        # not exist or is not unique. Same for the extension name. 
        for handler in ext_handlers:
            if handler_name == handler.name:
                resolved_dependencies.append(
                    Dependency(handler=handler,
                        exts = [ext in handler.properties.extensions where ext_name==ext.name]
                    ))
    self.dependencies = resolved_dependencies
```
I think this is more readable, but your mileage may vary.

Reply:
I'm a little twitchy in that a side effect of this function is to change the contents of self.dependencies from a list of 2-tuples to a list of Dependency objects. I know Python doesn't care, but does this alteration of the shape of that member make it harder for humans to understand what's going on?

Reply:
I would have put `Dependency` objects in the list from the get-go but I can't resolve the dependencies until it has already parsed all of the `Extension` objects because an extension's dependency might occur in the document after it has been parsed.

@boumenot thoughts on Jason's suggested list comprehension change?